lib/session/session.go

```diff
@@ -106,6 +106,8 @@ type Session struct {
 	AppName string `json:"app_name"`
 	// Owner is the name of the session owner, ie the one who created the session.
 	Owner string `json:"owner"`
+	// Moderated is true if the session requires moderation.
+	Moderated bool `json:"moderated"`
```

Comment:
Not directly related to your change but can we specify that this `Session` means **SSH** session in the docstring? We have lots of different types of "sessions" in the app and I sometimes get confused about what type of session (web session? ssh? k8s? etc) we're talking about when navigating the app.

```
// Session is an interactive collaboration SSH session that represents one
// or many SSH sessions started by the teleport user.
```

Reply:
I agree, however this `Session` has a `Kind` that looks like it can be any of 
```
SSHSessionKind            SessionKind            = "ssh"
KubernetesSessionKind     SessionKind            = "k8s"
DatabaseSessionKind       SessionKind            = "db"
AppSessionKind            SessionKind            = "app"
WindowsDesktopSessionKind SessionKind            = "desktop"
```

I'm not sure how that would translate. Although, when it comes to moderated, I believe we can only have a moderated session of ssh/k8s so we could say something like (correct me if I'm wrong). I'm not 100% sure we can specify SSH only in the doc string. But for `Moderated` docstring itself, maybe we say `if the ssh/k8s session requires moderated`.

Thoughts?

Reply:
Good point, I missed that. Now that I'm looking closer though, I'm actually only seeing `Kind` being either [hardcoded as SSH](https://github.com/gravitational/teleport/blob/9d8718cab88949e67923c958ca07d956a3b896e1/lib/web/apiserver.go#L2785), or else [set to a corresponding `Tracker`'s kind](https://github.com/gravitational/teleport/blob/9d8718cab88949e67923c958ca07d956a3b896e1/lib/web/apiserver.go#L2822), however afaict that latter function is only ever being called in contexts where we assume SSH session. Then there are places throughout the codebase where `Kind` remains unset, and afaict there we're assuming it to be an SSH session.

@kimlisa it looks like you added that field back in https://github.com/gravitational/teleport/issues/14329, is it still being used for that same purpose? If so, I can't find where. Perhaps we can now remove it and just assume that all such `Session` are SSH sessions.

Reply:
we need that field, the `session.Session` object is also used as the response object for web api calls. One is for getting a list of sessions and this is how it is used in front: https://github.com/gravitational/teleport/blob/25fe4222fcdd0435380b0ec63007baaaa8cbe11c/web/packages/teleport/src/Sessions/SessionList/SessionList.tsx#L86

Reply:
> or else [set to a corresponding Tracker's kind](https://github.com/gravitational/teleport/blob/9d8718cab88949e67923c958ca07d956a3b896e1/lib/web/apiserver.go#L2822), however afaict that latter function is only ever being called in contexts where we assume SSH session.

this was what i was referring to: https://github.com/gravitational/teleport/blob/25fe4222fcdd0435380b0ec63007baaaa8cbe11c/lib/web/apiserver.go#L2865 unless i'm misunderstanding, getting the list of active trackers can be of any kind

Reply:
@kimlisa yeah you're right, it's being used, but things are a bit confusing. @avatus go ahead and merge this (once you've responded to all the other feedback) and I'll make a separate PR cleaning this stuff up a bit.

Reply:
Follow up PR is here: https://github.com/gravitational/teleport/pull/24363